src/Sylius/Bundle/OrderBundle/DependencyInjection/Compiler/RegisterCartContextsPass.php

```diff
@@ -1,40 +0,0 @@
-<?php
-
-/*
- * This file is part of the Sylius package.
- *
- * (c) Paweł Jędrzejewski
- *
- * For the full copyright and license information, please view the LICENSE
- * file that was distributed with this source code.
- */
-
-declare(strict_types=1);
-
-namespace Sylius\Bundle\OrderBundle\DependencyInjection\Compiler;
-
-use Sylius\Bundle\ResourceBundle\DependencyInjection\Compiler\PrioritizedCompositeServicePass;
-use Sylius\Component\Order\Context\CartContextInterface;
-use Symfony\Component\DependencyInjection\ContainerBuilder;
-
-final class RegisterCartContextsPass extends PrioritizedCompositeServicePass
```

Comment:
I wonder, because after this refactoring, this class (and other compiler passes) won't make any sense at all, but removing it is a violation of BC policy IMO. Is there any chance, that someone could customize that logic and we will break them the end application, wdyt? 🤔

Reply:
We cannot know if someone used these classes in their projects, but because it's part of the public API it's covered by the BC promise, so we have to keep it together with the adders on the composite service classes.